source/jormungandr/jormungandr/olympic_site_params_manager.py

```diff
@@ -78,27 +116,44 @@ def get_dict_scenario(self, poi_uri, key):
         return self.build_olympic_site_params(data.get(key), data)
 
     def load_olympic_site_params(self, file_path, instance_name):
+
         logger = logging.getLogger(__name__)
         if not file_path:
-            logger.warning("Reading stop points attractivities, path does not found")
-            return
-        olympic_site_params_file = os.path.join(file_path, "{}.json".format(instance_name))
-        if not os.path.exists(olympic_site_params_file):
-            logger.warning(
-                "Reading stop points attractivities, file: %s does not exist", olympic_site_params_file
-            )
+            logger.debug("Reading stop points attractivities, path does not found")
             return
 
-        logger.info("Reading stop points attractivities from file: %s", olympic_site_params_file)
-        try:
-            with open(olympic_site_params_file) as f:
-                self.olympic_site_params = json.load(f)
-        except Exception as e:
-            logger.exception(
-                'Error while loading od_allowed_ids file: {} with exception: {}'.format(
-                    olympic_site_params_file, str(e)
+        json_pois_file = glob.glob(os.path.join(file_path, instance_name, "*.json"))
+        if not json_pois_file:
+            logger.debug("{} coverage without stop points attractivities".format(instance_name))
+            return
+        for json_poi_file in json_pois_file:
+            logger.debug("Reading stop points attractivities from file: %s", json_poi_file)
+            try:
+                with open(json_poi_file) as f:
+                    self.olympic_site_params.update(json.load(f))
+            except Exception as e:
+                logger.exception(
+                    'Error while loading od_allowed_ids file: {} with exception: {}'.format(
+                        json_poi_file, str(e)
+                    )
                 )
-            )
+
+    def build(self, pt_object_origin, pt_object_destination, api_request, instance):
+        api_request["olympic_site_params"] = self.get_olympic_site_params(
+            pt_object_origin, pt_object_destination, api_request, instance
+        )
+        self.add_criteria(api_request)
+
+    def add_criteria(self, api_request):
+        olympic_site_params = api_request.get("olympic_site_params")
+        if not olympic_site_params:
+            return
+        if api_request.get("criteria") in ["departure_stop_attractivity", "arrival_stop_attractivity"]:
+            return
+        if olympic_site_params.get("departure_scenario"):
+            api_request["criteria"] = "departure_stop_attractivity"
+        if olympic_site_params.get("arrival_scenario"):
```

Comment:
```suggestion
        elif olympic_site_params.get("arrival_scenario"):
```

Reply:
this part is strange... you mean, in case where both "departure_scenario" and "arrival_scenario" are present, you will take "arrival_stop_attractivity" as criteria?